src/linear_algebra/bilinear_form.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2018 Andreas Swerdlow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Andreas Swerdlow
+-/
+
+import linear_algebra.tensor_product
+
+universes u v
+
+structure bilin_form (α : Type u) (β : Type v) [ring α] [add_comm_group β] [module α β] :=
```

Comment:
Would life become easier if you just take the bilinear forms from `linear_algebra.tensor_product` as the definition? Then you would get lots of algebraic properties for free. Or does this lead to problems with type class search maxing out?

Reply:
As far as I can tell, it is useful to have bilinear forms on non-commutative rings, which I don't think is possible from the definition in `linear_algebra.tensor_product`